script/bootstrap

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env php
+# script/bootstrap: Resolve all dependencies that the application requires to
+#                   run.
+<?php
+error_reporting(E_ALL);
+
+// Go to LORIS root.
+chdir(dirname(__FILE__) . '/..');
+require('tools/PHP_CLI_Helper.class.inc');
+
+//TODO: Update these values as time passes 
+$required_major_php = 7;
+$required_minor_php = 2;
+$required_major_apache = 2;
+$required_minor_apache = 4;
+// PHP version required for LORIS.
+$required_php = "$required_major_php.$required_minor_php";
+
+/* Validate apache */
+// Get string representation of apache version number
+$apache_parts = explode(
+    '/', 
+    // this command yields e.g. Apache/2.4.34
+    shell_exec(
+        "apache2 -v | " . 
+        "head -n 1 | " .
+        "cut -d ' ' -f 3"
+    )
+);
+$apache_version = end($apache_parts);
+/* Look for the string "$major.$minor" in info string. Also match on versions
+ * higher than minor version because we want AT LEAST that version.
+ */
+$pattern = "/$required_major_apache\.[$required_minor_apache-9]/";
+if (preg_match($pattern, $apache_version) === 0) {
+    $required_apache = "$required_major_apache.$required_minor_apache";
+    die(
+        "ERROR: LORIS requires Apache v$required_apache or higher."
+        . PHP_EOL
+        . "This must be done manually as it has possible security ramifications."
+        . PHP_EOL
+    );
+}
+
+// NOTE Update as time passes.
+// Dependencies last updated for version: 20.0.1
+$apt_dependencies = [
+                       "wget",
+                       "zip",
+                       "unzip",
+                       "php-json",
+                       "npm",
+                       "software-properties-common",
+                       "php-ast",
+                       "php$required_php",
```

Comment:
depending on the distro, this might fail for the required version.

If the latest Ubuntu version directly support the required php version, it will simply be listed as "php". 
php -v is required

Reply:
Yes this is Ubuntu-specific as written. I don't have a CentOS VM to test on at the moment but if you do I'd appreciate some feedback about what needs to be added!

Ubuntu typically does not support the PHP version we need... it seems slow to get mixed into Debian/Ubuntu which is why adding the external PPA is baked-in here. 

`php -v` won't be sufficient as often the CLI PHP version differs from the version used in Apache. We aren't able to configure this for people necessarily as the Apache configuration might not follow the standards we use internally.

Reply:
CentOS (RHEL) tend to be ever slower than Ubuntu to catch up with the latest and greatest version of a package. 
I just wanted to be sure the script will always work on Ubuntu in the eventuality that Ubuntu (almost) catch up with the latest and greatest version of PHP.

And you are right about which version of PHP Apache use

Reply:
@johnsaigle 
FYI 
`rpm -q {package-name}`  will tell you if a package is installed
`rpm -qa` will list all packages that are installed

CentOS 7+ also have `dnf search {package-name}` but only for those installed using DNF (yum).